tests/cases/fourslash/completionsObjectLiteralMethod3.ts

```diff
@@ -0,0 +1,134 @@
+/// <reference path="fourslash.ts" />
+
+// @newline: LF
+// @Filename: a.ts
+////interface I1 {
+////    M(x: number): void;
+////}
+////interface I2 {
+////    M(x: number): void;
+////}
+////const u: I1 | I2 = {
+////    /*a*/
+////}
+////const i: I1 & I2 = {
+////    /*b*/
+////}
+////interface U1 {
+////    M(x: number): string;
+////}
+////interface U2 {
+////    M(x: string): number;
+////}
+////const o: U1 | U2 = {
+////    /*c*/
+////}
+////interface Op {
+////    M?(x: number): void;
+////    N: (x: string) => void | null | undefined;
+////    O?: () => void;
```

Comment:
I believe these optional ones are only working because the file is being checked without `strictNullChecks`. Also, `N` isn’t a union, it’s a function that returns a union 😄 

I think you just need to do the union type filtering you added _before_ checking the number of remaining call signatures, since `(() => any) | SomethingNonCallable` will return no signatures.

Reply:
> Also, N isn’t a union, it’s a function that returns a union 😄
🤦‍♀️thanks for noticing!